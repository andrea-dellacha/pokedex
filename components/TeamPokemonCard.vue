<template>
  <div class="cardPokemon">
    <div class="card__image">
      <img
        :id="`pokemon#${pokemon.id}`"
        :src="`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemon.id}.svg`"
        :alt="pokemon.name"
      />
    </div>
    <div class="card__title">
      <div class="card__name">{{ pokemon.name | capitalize }}</div>
      <div class="card__id">
        #{{ pokemon.id | formatPokemonId }}
        <div class="card__hp">
          <div class="cardHP__bar" />
          <div class="cardHP__text">{{ `${pokemon.hp} / ${pokemon.hp}` }}</div>
        </div>
      </div>
    </div>
    <div class="card__delete" @click="deletePokemon()">
      <TrashIcon />
    </div>
  </div>
</template>

<script>
import TrashIcon from '@/assets/icon/trash.vue'
export default {
  name: 'PokemonCard',
  components: {
    TrashIcon
  },
  props: {
    pokemon: {
      type: Object,
      required: true,
    },
  },
  methods: {
    deletePokemon () {
      this.$store.commit('team/deletePokemon', this.pokemon)
    }
  }
}
</script>

<style scoped>
.cardPokemon {
  width: 420px;
  background-color: white;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
  border-radius: 8px;
  display: grid;
  grid-template-columns: 90px 290px 30px;
  align-items: center;
  padding: 5px;
  color: black;
  cursor: move;
}

.card__image, .card__id {
  display: flex;
  justify-content: center;
  align-items: center;
}

.card__image img {
  max-width: 80px;
  max-height: 80px;
}

.card__title {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 5px;
  font-size: 24px;
  padding-right: 10px;
}

.card__name {
  white-space: nowrap;
}
.card__id {
  font-size: var(--font-size);
  padding-left: 10px;
  display: flex;
  justify-content: space-between;
  width: calc(100% - 10px);
}

.card__hp {
  display: flex;
  align-items: center;
  gap: 2px;

}

.cardHP__bar {
  height: 8px;
  border-radius: 5px;
  width: 100px;
  background-color: var(--green-color);
}

.cardHP__text {
  width: 70px;
}

.card__delete {
  display: grid;
  place-items: center;
  height: 100%;
  cursor: pointer;
}

@media screen and (max-width: 600px) {
  .card__title {
    font-size: 16px;
  }

  .cardPokemon {
    width: 250px;
    grid-template-columns: 30px 200px 20px;
  }

  .card__image img {
    max-width: 30px;
    max-height: 30px;
  }

  .cardHP__text {
    width: 55px;
  }

  .cardHP__bar {
    width: 75px;
  }

  .card__delete svg {
    width: 15px;
  }
}
</style>
