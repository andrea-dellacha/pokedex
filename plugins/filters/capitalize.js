import Vue from 'vue'

export default function capitalize (value) {
  if (!value) { return '' }
  value = value.toString()

  return value.charAt(0).toUpperCase() + value.slice(1)
}

Vue.filter('capitalize', capitalize)
