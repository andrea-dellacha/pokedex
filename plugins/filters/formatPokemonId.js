import Vue from 'vue'

export default function formatPokemonId (value) {
  if (value) {
    switch (value.toString().length) {
      case 1:
        return `00${value}`
      case 2:
        return `0${value}`
      case 3:
        return `${value}`
    }
  }  
}

Vue.filter('formatPokemonId', formatPokemonId)
