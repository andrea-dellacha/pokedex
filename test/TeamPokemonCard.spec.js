import { shallowMount, createLocalVue } from '@vue/test-utils'
import TeamPokemonCard from '@/components/TeamPokemonCard.vue'
import capitalize from '@/plugins/filters/capitalize'
import formatPokemonId from '@/plugins/filters/formatPokemonId'

describe('TeamPokemonCard', () => {
  const commit = jest.fn((n) => {})
  const localVue = createLocalVue()

  // Filter import
  localVue.filter('capitalize', capitalize)
  localVue.filter('formatPokemonId', formatPokemonId)

  const wrapper = shallowMount(TeamPokemonCard, {
    mocks: {
      $store: {
        commit
      }
    },
    propsData: {
      pokemon: {
        id: 5,
        name: 'charmander'
      }
    }
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should commit deletePokemon', () => {
    commit.mockReset()
    wrapper.vm.deletePokemon()
    expect(commit).toHaveBeenCalledWith('team/deletePokemon', wrapper.vm.pokemon)
  })
})
