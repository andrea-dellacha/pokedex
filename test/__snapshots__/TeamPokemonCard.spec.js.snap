// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPokemonCard renders correctly 1`] = `
<div
  class="cardPokemon"
>
  <div
    class="card__image"
  >
    <img
      alt="charmander"
      id="pokemon#5"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg"
    />
  </div>
   
  <div
    class="card__title"
  >
    <div
      class="card__name"
    >
      Charmander
    </div>
     
    <div
      class="card__id"
    >
      
      #005
      
      <div
        class="card__hp"
      >
        <div
          class="cardHP__bar"
        />
         
        <div
          class="cardHP__text"
        >
          undefined / undefined
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="card__delete"
  >
    <trashicon-stub />
  </div>
</div>
`;
